src/components/manage_dao/edit_form.tsx

```diff
@@ -1,172 +1,328 @@
+								toast.close(toastRef.current)
+							}
+						}
+					}
+				/>
+			),
+		})
+	}
+
+	const updateEditError = (
+		key: keyof SettingsForm,
+		error: string | undefined
+	) => setEditError((err) => ({ ...err, [key]: error ? { error } : undefined }))
+
+	const updateFormData = (update: Partial<SettingsForm>) => {
+		// eslint-disable-next-line guard-for-in, no-restricted-syntax
+		for(const key in update) {
+			updateEditError(key as any, undefined)
+		}
+
+		if(editedFormData) {
+			setEditedFormData((current) => ({ ...current!, ...update }))
 		}
+	}
 
-		const supportedChainId = getSupportedChainIdFromSupportedNetwork(formData.supportedNetwork)
-		const networkName = supportedChainId ? ((CHAIN_INFO[supportedChainId].isTestNetwork && !SHOW_TEST_NETS) ? 'Unsupported Network' : CHAIN_INFO[supportedChainId].name) : 'Unsupported Network'
-		setDaoName(formData.name)
-		setDaoAbout(formData.about)
-		setSupportedNetwork(networkName)
-		setImage(formData.image)
-		setCoverImage(formData.coverImage)
-		setTwitterHandle(formData.twitterHandle)
-		setDiscordHandle(formData.discordHandle)
-		setTelegramChannel(formData.telegramChannel)
-	}, [formData])
-
-	const handleImageChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+	const hasError = (key: keyof SettingsForm) => !!editError[key]
+
+	const handleImageChange = (
+		key: 'image' | 'coverImage',
+		event: React.ChangeEvent<HTMLInputElement>
+	) => {
 		if(event.target.files && event.target.files[0]) {
 			const img = event.target.files[0]
-			setImageFile(img)
-			setImage(URL.createObjectURL(img))
+			updateFormData({ [key]: URL.createObjectURL(img).toString() })
 		}
 	}
 
-	const handleCoverImageChange = (
+	const handlePartnerImageChange = (
 		event: React.ChangeEvent<HTMLInputElement>,
+		index: number
 	) => {
+
 		if(event.target.files && event.target.files[0]) {
 			const img = event.target.files[0]
-			setCoverImageFile(img)
-			setCoverImage(URL.createObjectURL(img))
+			const maxImageSize = 2
+			if(img.size / 1024 / 1024 <= maxImageSize) {
+				const oldPartners = [...partners!]
+				oldPartners[index].image = URL.createObjectURL(img)
+				setPartners(oldPartners)
+				setChangedPartners(true)
+			} else {
+				toastRef.current = toast({
+					position: 'top',
+					render: () => ErrorToast({
+						content: `Image size exceeds ${maxImageSize} MB`,
+						close: () => {
+							if(toastRef.current) {
+								toast.close(toastRef.current)
+							}
+						},
+					}),
+				})
+			}
 		}
 	}
 
-	const handleSubmit = () => {
-		let error = false
-		if(!daoName || daoName.length === 0) {
-			setDaoNameError(true)
-			error = true
+	const handleSubmit = async() => {
+		if(changedAbout) {
+			const newAboutString = await JSON.stringify(
+				convertToRaw(newAbout.getCurrentContent())
+			)
+
+			const newAboutHash = (await uploadToIPFS(newAboutString)).hash
+			updateFormData({ about: newAboutHash })
+		}
+
+		if(changedPartners || changedPartnersImage) {
```

Comment:
should put this at the end after everything has been validated, will return in unnecessary requests